client/device/cloudfmc/create.go

```diff
@@ -0,0 +1,121 @@
+package cloudfmc
+
+import (
+	"context"
+	"errors"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/device"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/device/application"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/goutil"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/retry"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/application/applicationstatus"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/model/devicetype"
+	"time"
+)
+
+type CreateInput struct {
+}
+
+func NewCreateInput() CreateInput {
+	return CreateInput{}
+}
+
+type createApplicationBody struct {
+	ApplicationType    string             `json:"applicationType"`
+	ApplicationStatus  string             `json:"applicationStatus"`
+	ApplicationContent applicationContent `json:"applicationContent"`
+}
+
+type applicationContent struct {
+	Type string `json:"@type"`
+}
+
+type CreateOutput = device.CreateOutput
+
+func Create(ctx context.Context, client http.Client, createInp CreateInput) (*CreateOutput, error) {
+
+	client.Logger.Println("creating cloud FMC")
+
+	client.Logger.Println("creating cloud FMC application")
+
+	// 1. POST /aegis/rest/v1/services/targets/applications
+	createApplicationUrl := url.CreateApplication(client.BaseUrl())
+	createApplicationReq := client.NewPost(
+		ctx,
+		createApplicationUrl,
+		createApplicationBody{
+			ApplicationType:   "FMCE",
+			ApplicationStatus: "REQUESTED",
+			ApplicationContent: applicationContent{
+				Type: "FmceApplicationContent",
+			},
+		},
+	)
+	var createApplicationOutp device.CreateOutput
+	err := createApplicationReq.Send(&createApplicationOutp)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Logger.Println("creating cloud FMC device")
+
+	// 2. POST /aegis/rest/v1/services/targets/devices
+	createOutp, err := device.Create(ctx, client, device.NewCreateInputBuilder().
+		Name("FMC").
+		DeviceType(devicetype.CloudFmc).
+		Model(false).
+		ConnectorType("CDG").
+		IgnoreCertificate(goutil.NewBoolPointer(true)).
+		EnableOobDetection(goutil.NewBoolPointer(false)).
+		Build(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	client.Logger.Println("waiting for fmce state machine to be done")
+
+	err = retry.Do(untilApplicationActive(ctx, client),
+		retry.NewOptionsBuilder().
+			Retries(-1).
+			Timeout(time.Hour). // usually takes about 15-20 minutes
```

Comment:
I think we need to reduce the timeout here to 30 mins, if it's processing longer then it probably causes errors

Reply:
make sense, done 👍